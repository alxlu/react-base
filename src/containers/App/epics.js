import { delay } from 'rxjs/operator/delay';
import { mapTo } from 'rxjs/operator/mapTo';
import { filter } from 'rxjs/operator/filter';
import { LOCATION_CHANGE } from 'react-router-redux';

const appEpic = (action$, store) => (
  action$::filter(action => action.type === LOCATION_CHANGE)
    ::delay(1000)
    ::mapTo({ type: 'PONG' })
);

export default appEpic;
