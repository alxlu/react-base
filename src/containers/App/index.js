import React from 'react';
import { Link } from 'react-router';

type Props = {
  children: Object;
};

const App = (props: Props) => (
  <div>
    <h1>App Homepage</h1>
    <Link to="/">Home</Link>
    <Link to="/foo">Foo</Link>
    <div>{props.children}</div>
  </div>
);

export default App;
