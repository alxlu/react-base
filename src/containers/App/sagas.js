export default function* appSaga() {
  // saga
}
