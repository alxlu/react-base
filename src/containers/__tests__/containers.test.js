/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
import Home from '../Home';
import TestPage from '../TestPage';

describe('containers', () => {
  describe('component rendering', () => {
    it('<App /> renders correctly', () => {
      const tree = renderer.create(
        <App />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('<Home /> renders correctly', () => {
      const tree = renderer.create(
        <Home />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('<TestPage /> renders correctly', () => {
      const tree = renderer.create(
        <TestPage />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
