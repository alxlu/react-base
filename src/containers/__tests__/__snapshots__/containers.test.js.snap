// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers component rendering <App /> renders correctly 1`] = `
<div>
  <h1>
    App Homepage
  </h1>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    Home
  </a>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    Foo
  </a>
  <div />
</div>
`;

exports[`containers component rendering <Home /> renders correctly 1`] = `
<div>
  Home
</div>
`;

exports[`containers component rendering <TestPage /> renders correctly 1`] = `
<div>
  Test
</div>
`;
